### ISSUE TITLE
Seaweed crashes while vacuuming (invalid memory address or nil pointer dereference)

### ISSUE THREAD

**Issue by MrBoombastic** [seaweedfs/seaweedfs#4665]
**Describe the bug**
Seaweed crashes while vacuuming a volume. However, everything works fine while using older version (`volume_grpc_vacuum.go:87 failed commit volume 2: remove files/2.dat: The process cannot access the file because it is being used by another process.`) Is that a regression bug? Also, I don't know which process blocks that file.

**System Setup**
- `weed.exe server -dir=files/ -s3`
- Windows 11 Build 22631.1972
- `version 30GB 3.53 2c4c2f0994604510631920f4d0d9ee817ec29224 windows amd64`

**Expected behavior**
Seaweed should survive. 😄

**Logs**
Logs for both 3.52 and 3.53: https://gist.github.com/MrBoombastic/91e4573176aa376ef3a86b9eca133c7b

**Comment by chrislusf** [comment]
Is this always reproducible?

The log showed it's an error with the grpc library 1.55.0.
The latest code is using `google.golang.org/grpc v1.56.2`

**Comment by MrBoombastic** [comment]
Yes, I can alway reproduce this error while using v3.53 - I just have to wait 15 minutes. I'm using binaries from Releases tab. However, v3.54 works perfectly fine, so I guess this issue is already fixed (Seaweed doesn't crash), but I sitll don't know why it can't access `files/2.dat` file.

Fresh logs for both versions: https://gist.github.com/MrBoombastic/679aca8799074b954f33ccb85eb78e7c
